From b525767c86cd90b21d5079a4642f01b64941fd79 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Wed, 23 Feb 2022 22:44:28 +0100
Subject: [PATCH] startkde: Forward stdout/stderr of started processes

QProcess::start defaults to capturing all output, but that got lost as it is
never read. Set QProcess::ForwardedChannels explicitly where it was missing,
so that output ends up in the correct place again.

(cherry picked from commit d1155030afb5f94fc81e7afc4d1480d21d7d73f9)
---
 startkde/plasma-session/startup.cpp | 2 ++
 startkde/startplasma.cpp            | 1 +
 2 files changed, 3 insertions(+)

diff --git a/startkde/plasma-session/startup.cpp b/startkde/plasma-session/startup.cpp
index 246d88b35..9078c1f16 100644
--- a/startkde/plasma-session/startup.cpp
+++ b/startkde/plasma-session/startup.cpp
@@ -246,6 +246,7 @@ bool Startup::startDetached(const QString &program, const QStringList &args)
 
 bool Startup::startDetached(QProcess *process)
 {
+    process->setProcessChannelMode(QProcess::ForwardedChannels);
     process->start();
     const bool ret = process->waitForStarted();
     if (ret) {
@@ -428,6 +429,7 @@ StartProcessJob::StartProcessJob(const QString &process, const QStringList &args
 {
     m_process->setProgram(process);
     m_process->setArguments(args);
+    m_process->setProcessChannelMode(QProcess::ForwardedChannels);
     auto env = QProcessEnvironment::systemEnvironment();
     env.insert(additionalEnv);
     m_process->setProcessEnvironment(env);
diff --git a/startkde/startplasma.cpp b/startkde/startplasma.cpp
index a68e7c023..bd848dda0 100644
--- a/startkde/startplasma.cpp
+++ b/startkde/startplasma.cpp
@@ -500,6 +500,7 @@ QProcess *setupKSplash()
         KConfigGroup ksplashCfg = cfg.group("KSplash");
         if (ksplashCfg.readEntry("Engine", QStringLiteral("KSplashQML")) == QLatin1String("KSplashQML")) {
             p = new QProcess;
+            p->setProcessChannelMode(QProcess::ForwardedChannels);
             p->start(QStringLiteral("ksplashqml"), {ksplashCfg.readEntry("Theme", QStringLiteral("Breeze"))});
         }
     }
-- 
2.34.0

